### PR TITLE
Fix TEE download error when agent and work directory are located on different mounts

### PR DESCRIPTION
**Pull request by DaniilShmelev**
`Directory.Move` fails when trying to move directory between mounts. This is a known and documented issue in dotnet: https://github.com/dotnet/runtime/issues/31149
This PR replaces `Directory.Move` call with a recursive copy (`IOUtil.CopyDirectory`). This should prevent failures for this case.
Related issue: https://github.com/microsoft/azure-pipelines-agent/issues/3761

### CODE REVIEW

**Review by KonstantinTyukalov — APPROVED** [review]

**Comment by anatolybolshakov** [comment]
@DaniilShmelev thanks! Could you please cover this by tests also?

**Review by alexander-smolyakov — APPROVED** [review]

**Comment by DaniilShmelev** [comment]
@anatolybolshakov I've created a separate item for enhancement of this feature. I'll add unit tests as a part of it.

**Review by KonstantinTyukalov — APPROVED** [review]

File: src/Agent.Sdk/Util/TeeUtil.cs [hunk 1]
```diff
@@ -83,12 +83,12 @@ CancellationToken cancellationToken
             string extractedTeePath = Path.Combine(tempDirectory, $"{Guid.NewGuid().ToString()}");
             ZipFile.ExtractToDirectory(zipPath, extractedTeePath);
 
-            debug($"Extracted {zipPath} to ${extractedTeePath}");
+            debug($"Extracted {zipPath} to {extractedTeePath}");
 
             string extractedTeeDestinationPath = GetTeePath();
-            Directory.Move(Path.Combine(extractedTeePath, TeePluginName), extractedTeeDestinationPath);
+            IOUtil.CopyDirectory(Path.Combine(extractedTeePath, TeePluginName), extractedTeeDestinationPath, cancellationToken);
```
Comment (anatolybolshakov) [inline src/Agent.Sdk/Util/TeeUtil.cs:89]: Should we remove extractedTeePath (since we moved it prevously)?

Comment (DaniilShmelev) [reply src/Agent.Sdk/Util/TeeUtil.cs:89]: extractedTeePath is located inside tempDirectory, and we delete tempDirectory a couple of lines later

**Review by mmrazik — APPROVED** [review]

**Review by DenisRumyantsev — APPROVED** [review]
Tested these changes end-to-end manually, agent works properly, no any concerns, approved.

**Review by anatolybolshakov — APPROVED** [review]

**Pull request merged**